### PR TITLE
Fix frontend coverage and tweak ignores

### LINKED ISSUE #2552: Coverage is misreported for the frontend
Some of the files aren't being reported:

screen shot 2018-12-18 at 4 24 20 pm

Maybe it's transpiler-related?

### PR DESCRIPTION
**Pull request by paulmelnikow**
Since we use Babel for the frontend but not the server, I split the coverage config into two separate files, one for the frontend and the other for the server.

Refs:

- https://github.com/istanbuljs/nyc#nyc
- https://github.com/istanbuljs/babel-plugin-istanbul#mocha-on-nodejs-through-nyc

Close #2552

### CODE REVIEW

**Comment by paulmelnikow** [comment]
Can't see this in coveralls, though in a local run of most of the tests, the coverage report looks good.

File: .nycrc.json [hunk 1]
```diff
@@ -1,17 +1,22 @@
 {
   "reporter": ["lcov"],
```
Comment (calebcartwright) [inline .nycrc.json]: what do folks think about adding a `text` and/or `text-summary` reporter for pretty printing of coverage in the terminal ? (unless that is already available and I just haven't found it yet 😄 )

File: .nycrc.json [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+  "silent": true,
+  "clean": false,
   "exclude": [
     "**/*.spec.js",
     "**/*.integration.js",
     "dangerfile.js",
     "next.config.js",
     "services/**/*.tester.js",
+    "services/service-tester.js",
+    "services/create-service-tester.js",
```
Comment (calebcartwright) [inline .nycrc.json]: 👍

File: .nycrc.json (hunk 1 above)
Comment (paulmelnikow) [reply .nycrc.json]: Sure! The only thing which makes our setup a little tricky is that there are four separate tests runs which all contribute to coverage. That's why we use `clean: true`. I'm not sure how that might interact with the terminal reporters.

Want to push some more commits to this branch?

Comment (calebcartwright) [reply .nycrc.json]: Ah yes I can do that now can't I!

File: package.json [hunk 3]
```diff
@@ -58,7 +58,7 @@
     "xpath": "~0.0.27"
   },
   "scripts": {
-    "coverage:test:frontend": "BABEL_ENV=test nyc node_modules/mocha/bin/_mocha --opts mocha.opts --require @babel/polyfill --require @babel/register --require mocha-yaml-loader \"frontend/**/*.spec.js\"",
+    "coverage:test:frontend": "BABEL_ENV=test nyc --nycrc-path .nycrc-frontend.json node_modules/mocha/bin/_mocha --opts mocha.opts --require @babel/polyfill --require @babel/register --require mocha-yaml-loader \"frontend/**/*.spec.js\"",
     "coverage:test:server": "HANDLE_INTERNAL_ERRORS=false nyc node_modules/mocha/bin/_mocha --opts mocha.opts \"*.spec.js\" \"lib/**/*.spec.js\" \"services/**/*.spec.js\"",
```
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply package.json]: I'm not completely sure. The `nyc` docs say you can use `mocha`. At one point that was not the case. I think the reason may have related to command-line arguments, though I honestly don't remember.

I found gotwarlost/istanbul#436 which is not very helpful; mentions https://github.com/gotwarlost/istanbul/issues/44#issuecomment-16093330 which indeed doesn't explain it.

I haven't tried swapping it back out, though if it works with `mocha`, let's change it.

**Comment by paulmelnikow** [comment]
Nice!

**Comment by paulmelnikow** [comment]
This doesn't seem to be working right in the daily tests.

**Comment by calebcartwright** [comment]
> This doesn't seem to be working right in the daily tests.

I'm seeing this in the Circle logs for the daily tests:

```
> shields.io@0.0.0 coverage:test:frontend /home/circleci/project/shields
> BABEL_ENV=test nyc --nycrc-path .nycrc-frontend.json node_modules/mocha/bin/_mocha --opts mocha.opts --require @babel/polyfill --require @babel/register --require mocha-yaml-loader "frontend/**/*.spec.js"

module.js:550
    throw err;
    ^

Error: Cannot find module '../../supported-features.json'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/circleci/project/shields/frontend/components/usage.js:5:1)
    at Module._compile (module.js:653:30)
    at Module.replacementCompile (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at Module._compile (/home/circleci/project/shields/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (module.js:664:10)
    at /home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:62:4
    at newLoader (/home/circleci/project/shields/node_modules/pirates/lib/index.js:88:7)
    at Object.<anonymous> (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/circleci/project/shields/frontend/components/usage.spec.js:4:1)
    at Module._compile (module.js:653:30)
    at Module.replacementCompile (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at Module._compile (/home/circleci/project/shields/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (module.js:664:10)
    at /home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:62:4
    at newLoader (/home/circleci/project/shields/node_modules/pirates/lib/index.js:88:7)
    at Object.<anonymous> (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at /home/circleci/project/shields/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/circleci/project/shields/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/home/circleci/project/shields/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/home/circleci/project/shields/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (module.js:653:30)
    at Module.replacementCompile (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at Module._extensions..js (module.js:664:10)
    at Object.<anonymous> (/home/circleci/project/shields/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at runMain (/home/circleci/.node-spawn-wrap-1113-f5882b591bdb/node:68:10)
    at Function.<anonymous> (/home/circleci/.node-spawn-wrap-1113-f5882b591bdb/node:171:5)
    at Object.<anonymous> (/home/circleci/project/shields/node_modules/nyc/bin/wrap.js:23:4)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at /home/circleci/.node-spawn-wrap-1113-f5882b591bdb/node:178:8
    at Object.<anonymous> (/home/circleci/.node-spawn-wrap-1113-f5882b591bdb/node:181:3)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! shields.io@0.0.0 coverage:test:frontend: `BABEL_ENV=test nyc --nycrc-path .nycrc-frontend.json node_modules/mocha/bin/_mocha --opts mocha.opts --require @babel/polyfill --require @babel/register --require mocha-yaml-loader "frontend/**/*.spec.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the shields.io@0.0.0 coverage:test:frontend script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2018-12-20T03_45_46_611Z-debug.log
```

**Comment by paulmelnikow** [comment]
Aha!

**Pull request merged**